xdsl/dialects/riscv.py

```diff
@@ -496,28 +495,18 @@ def assembly_line(self) -> str | None:
 
 
 def _assembly_arg_str(arg: AssemblyInstructionArg) -> str:
-    if isa(arg, IntegerAttr):
+    if isinstance(arg, SSAValue):
+        if not isinstance(t := arg.type, RegisterType):
+            raise ValueError(f"Unexpected register type {t}")
+        return t.register_name.data
+    elif isinstance(arg, IntegerAttr):
         return f"{arg.value.data}"
```

Comment:
unrelated to this PR, but we should really consider using `printer.print_int` here if that's an option I feel